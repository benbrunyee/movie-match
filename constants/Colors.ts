const tintColorLight = "#2f95dc";
const tintColorDark = "#fff";

export default {
  light: {
    text: "#000",
    textDisabled: "#bbb",
    placeholderText: "#ddd",
    background: "#f3f3f3",
    containerBackgroundDefault: "#fff",
    containerBackgroundPressed: "#eee",
    tint: tintColorLight,
    tabIconDefault: "#ccc",
    tabIconSelected: tintColorLight,
    tabBackgroundDefault: "#eee",
    tabBackgroundSelected: "#ddd",
    tabBackgroundPressed: "#ddd",
    buttonBackgroundDefault: "#eee",
    buttonBackgroundPressed: "#ccc",
    borderColor: "#ddd",
  },
  dark: {
    text: "#fff",
    textDisabled: "#aaa",
    placeholderText: "#ddd",
    background: "#222",
    containerBackgroundDefault: "#666",
    containerBackgroundPressed: "#888",
    tint: tintColorDark,
    tabIconDefault: "#ccc",
    tabIconSelected: tintColorDark,
    tabBackgroundDefault: "#666",
    tabBackgroundSelected: "#444",
    tabBackgroundPressed: "#444",
    buttonBackgroundDefault: "#666",
    buttonBackgroundPressed: "#888",
    borderColor: "#444",
  },
};
