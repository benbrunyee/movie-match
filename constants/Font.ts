export const sizes = {
  title: 22,
  subtitle: 20,
  body: 18,
  caption: 16,
};
