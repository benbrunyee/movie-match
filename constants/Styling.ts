export default {
  spacingSmall: 10,
  spacingMedium: 20,
  spacingLarge: 40,
  borderRadius: 5,
};