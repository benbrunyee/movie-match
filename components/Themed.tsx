/**
 * Learn more about Light and Dark modes:
 * https://docs.expo.io/guides/color-schemes/
 */

import {
  Pressable,
  PressableProps,
  Text as DefaultText,
  TextInput as DefaultTextInput,
  View as DefaultView
} from "react-native";
import Colors from "../constants/Colors";
import { sizes } from "../constants/Font";
import Styling from "../constants/Styling";
import useColorScheme from "../hooks/useColorScheme";

export function useThemeColor(
  props: { light?: string; dark?: string },
  colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) {
  const theme = useColorScheme();
  const colorFromProps = props[theme];

  return colorFromProps || Colors[theme][colorName];
}

export function useFontSize(variant?: keyof typeof sizes) {
  return variant ? sizes[variant] : sizes.body;
}

type ThemeProps = {
  lightColor?: string;
  darkColor?: string;
};

type FontProps = {
  variant?: keyof typeof sizes;
};

export type TextProps = ThemeProps &
  FontProps &
  DefaultText["props"] & { disabled?: boolean };
export type BoxProps = ThemeProps & DefaultView["props"];
export type TextInputProps = ThemeProps & FontProps & DefaultTextInput["props"];
export type ButtonProps = ThemeProps & PressableProps;
export type TabProps = ThemeProps & { selected?: boolean } & PressableProps;
export type ContainerProps = ThemeProps & DefaultView["props"];
export type MenuItemProps = ThemeProps &
  PressableProps & { topBorder?: boolean; bottomBorder?: boolean };

export function Text(props: TextProps) {
  const { disabled, variant, style, lightColor, darkColor, ...otherProps } =
    props;
  const color = useThemeColor(
    { light: lightColor, dark: darkColor },
    disabled ? "textDisabled" : "text"
  );
  const fontSize = useFontSize(variant);

  return <DefaultText style={[{ color, fontSize }, style]} {...otherProps} />;
}

export function Box(props: BoxProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "background"
  );

  return <DefaultView style={[{ backgroundColor }, style]} {...otherProps} />;
}

export function TextInput(props: TextInputProps) {
  const { variant, style, lightColor, darkColor, ...otherProps } = props;
  const color = useThemeColor({ light: lightColor, dark: darkColor }, "text");
  const placeholderTextColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "placeholderText"
  );
  const fontSize = useFontSize(variant);

  return (
    <DefaultTextInput
      placeholderTextColor={placeholderTextColor}
      style={[{ color, fontSize, margin: Styling.spacingSmall }, style]}
      {...otherProps}
    />
  );
}

export function Button(props: ButtonProps) {
  const { disabled, style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "buttonBackgroundDefault"
  );
  const backgroundColorPressed = useThemeColor(
    { light: lightColor, dark: darkColor },
    "buttonBackgroundPressed"
  );

  return (
    <Pressable
      style={({ pressed }) => [
        {
          backgroundColor: pressed ? backgroundColorPressed : backgroundColor,
          paddingVertical: Styling.spacingSmall,
          paddingHorizontal: Styling.spacingMedium,
          borderRadius: Styling.borderRadius,
        },
        typeof style === "function" ? style({ pressed }) : style,
      ]}
      {...otherProps}
    />
  );
}

export function Tab(props: TabProps) {
  const { onPress, selected, style, lightColor, darkColor, ...otherProps } =
    props;
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    selected ? "tabBackgroundSelected" : "tabBackgroundDefault"
  );
  const pressedBackgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "tabBackgroundPressed"
  );

  return (
    <Pressable
      onPress={onPress}
      style={({ pressed }) => [
        {
          backgroundColor: pressed ? pressedBackgroundColor : backgroundColor,
        },
        typeof style === "function" ? style({ pressed }) : style,
      ]}
      {...otherProps}
    />
  );
}

export function Container(props: ContainerProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "containerBackgroundDefault"
  );

  return (
    <DefaultView
      style={[{ backgroundColor, borderRadius: Styling.borderRadius }, style]}
      {...otherProps}
    />
  );
}

export function MenuItem(props: MenuItemProps) {
  const {
    bottomBorder,
    topBorder,
    lightColor,
    darkColor,
    style,
    ...otherProps
  } = props;
  const backgroundColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "containerBackgroundDefault"
  );
  const backgroundColorPressed = useThemeColor(
    { light: lightColor, dark: darkColor },
    "containerBackgroundPressed"
  );
  const borderColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "borderColor"
  );

  return (
    <Pressable
      style={({ pressed }) => [
        {
          backgroundColor: pressed ? backgroundColorPressed : backgroundColor,
          ...(bottomBorder && {
            borderBottomWidth: 1,
            borderBottomColor: borderColor,
          }),
          ...(topBorder && { borderTopWidth: 1, borderTopColor: borderColor }),
          padding: Styling.spacingMedium,
        },
        typeof style === "function" ? style({ pressed }) : style,
      ]}
      {...otherProps}
    />
  );
}
