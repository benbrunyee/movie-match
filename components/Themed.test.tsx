// @ts-nocheck

import React from "react";
import renderer from "react-test-renderer";
import Colors from "../constants/Colors";
import { sizes } from "../constants/Font";
import { Button, Text, TextInput } from "./Themed";

describe("<Text />", () => {
  const text = "Hello world!";

  it("renders correctly", () => {
    const tree = renderer.create(<Text>{text}</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("has body font size as default", () => {
    const tree = renderer.create(<Text>{text}</Text>).toJSON();
    expect(tree.props.style[0]).toMatchObject({ fontSize: sizes.body });
  });
  it("renders font sizes correctly", () => {
    const smallCaption = renderer
      .create(<Text variant="smallCaption">{text}</Text>)
      .toJSON();
    const caption = renderer
      .create(<Text variant="caption">{text}</Text>)
      .toJSON();
    const body = renderer.create(<Text variant="body">{text}</Text>).toJSON();
    const subtitle = renderer
      .create(<Text variant="subtitle">{text}</Text>)
      .toJSON();
    const title = renderer.create(<Text variant="title">{text}</Text>).toJSON();

    expect(smallCaption.props.style[0]).toMatchObject({
      fontSize: sizes.smallCaption,
    });
    expect(caption.props.style[0]).toMatchObject({ fontSize: sizes.caption });
    expect(body.props.style[0]).toMatchObject({ fontSize: sizes.body });
    expect(subtitle.props.style[0]).toMatchObject({ fontSize: sizes.subtitle });
    expect(title.props.style[0]).toMatchObject({ fontSize: sizes.title });
  });
  it("matches disabled color when disabled", () => {
    const tree = renderer.create(<Text disabled>{text}</Text>).toJSON();
    expect(tree.props.style[0]).toMatchObject({
      color: Colors.light.textDisabled,
    });
  });
});

describe("<TextInput />", () => {
  const input = "Hello World!";

  it("renders correctly", () => {
    const tree = renderer.create(<TextInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("can take input", () => {
    const tree = renderer.create(<TextInput value={input} />).toJSON();
    expect(tree.props.value).toBe(input);

    const newInput = "New Input";
    tree.props.value = newInput;

    expect(tree.props.value).toBe(newInput);
  });
  it("has placeholder text color", () => {
    const tree = renderer.create(<TextInput placeholder={input} />).toJSON();
    expect(tree.props.placeholderTextColor).toBe(Colors.light.placeholderText);
  });
  it("has default body font size", () => {
    const tree = renderer.create(<TextInput value={input} />).toJSON();
    expect(tree.props.style[0]).toMatchObject({ fontSize: sizes.body });
  });
});

describe("<Button />", () => {
  const text = "Hello World!";

  it("renders correctly", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
