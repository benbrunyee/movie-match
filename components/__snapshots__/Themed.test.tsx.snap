// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#eee",
        "borderRadius": 5,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
/>
`;

exports[`<Text /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontSize": 18,
      },
      undefined,
    ]
  }
>
  Hello world!
</Text>
`;

exports[`<TextInput /> renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  placeholderTextColor="#ddd"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "color": "#000",
        "fontSize": 18,
        "margin": 10,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
